lab_3_ann_retriever/main.py

```diff
@@ -220,6 +299,14 @@ def _calculate_tf_idf(self, document: list[str]) -> Vector | None:
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(document, list) or not all(isinstance(token, str) for token in document) \
+                or not document:
+            return None
+
+        self.build()
```

Comment:
why are you bulding in calculate??

Reply:
so, may be tests are irrelevant or you have mistake in another place. Check it, please

Reply:
finally managed to fix it!

Reply:
NICE!!!